packages/hash/api/src/graphql/resolvers/taskExecutor/index.ts

```diff
@@ -22,3 +32,217 @@ export const executeDemoTask: Resolver<
     }
   }
 };
+
+export const executeGithubSpecTask: Resolver<
+  Promise<string>,
+  {},
+  GraphQLContext
+> = async (_, __, { dataSources: { taskExecutor } }) => {
+  if (!taskExecutor) {
+    throw new ApolloError(
+      "A task-executor wasn't started, so external tasks can't be started",
+    );
+  } else {
+    try {
+      return await taskExecutor
+        .run_task(Tasks.GithubSpec)
+        .then((res) => JSON.stringify(res));
+    } catch (err: any) {
+      throw new ApolloError(`Task-execution failed: ${err}`);
+    }
+  }
+};
+
+export const executeGithubCheckTask: Resolver<
+  Promise<string>,
+  {},
+  GraphQLContext,
+  MutationExecuteGithubCheckTaskArgs
+> = async (_, { config }, { dataSources: { taskExecutor } }) => {
+  if (!taskExecutor) {
+    throw new ApolloError(
+      "A task-executor wasn't started, so external tasks can't be started",
+    );
+  } else {
+    try {
+      return await taskExecutor
+        .run_task(Tasks.GithubCheck, config)
+        .then((res) => JSON.stringify(res));
+    } catch (err: any) {
+      throw new ApolloError(`Task-execution failed: ${err}`);
+    }
+  }
+};
+
+/** @todo - Make Airbyte types available in api package */
+type AirbyteCatalog = Array<{
+  name?: string;
+  json_schema?: JSONObject;
+}>;
+
+type AirbyteRecords = Array<{
+  /**
+   * the name of this record's stream
+   */
+  stream: string;
+  /**
+   * the record data
+   */
+  data: {
+    [k: string]: unknown;
+  };
+  /**
+   * when the data was emitted from the source. epoch in millisecond.
+   */
+  emitted_at: number;
+  /**
+   * the namespace of this record's stream
+   */
+  namespace?: string;
+  [k: string]: unknown;
+}>;
+
+const ExistingEntityChecker = async (db: DbAdapter, user: User) => {
```

Comment:
Agreed on that too, this was mostly a hack to get my development forwards, and I opened the other comment here to check what we want to do with it, wasn't expecting us to merge it as is. I suspect we instead want to actually add a method on the EntityType class (just realised this should be `ExistingEntityTypeChecker` anyway), _perhaps_ even before merging this PR

Reply:
Do we want to add a `getByAccountIdAndName` method or something on the `EntityType` class?

Reply:
Sorry, didn't see this response -
I think it might take a little more work, as the return value of this function is a little unconventional compared to the rest of the methods in the model class. 

While the consistency would be nice, I think moving it to `api/src/task-executor` would be fine for now, and then we can move it into the model class down the line. For me the thin resolvers are more important in this case

Reply:
Makes sense, moved it